lib/config-generator.js

```diff
@@ -320,20 +320,26 @@ class ConfigGenerator {
          *  * https://github.com/webpack/webpack.js.org/issues/652#issuecomment-273324529
          *  * https://webpack.js.org/guides/caching/#deterministic-hashes
          */
-        // when not using versioning, none of this is important
-        if (this.webpackConfig.useVersioning) {
-            let moduleNamePlugin;
-            if (this.webpackConfig.isProduction()) {
+        let moduleNamePlugin;
+        if (this.webpackConfig.isProduction()) {
+            // only enable if versioning is on, as it makes the assets *slightly* larger
```

Comment:
@weaverryan my issue is stable module names, not stable output files.

hashed-asset-bundle uses a (truncated) hash of the file content as version. So if you use numeric index for modules, the hash of the vendor module may change just because of reindexing module, busting the cache for nothing.

you use the HashedModuleIdsPlugin when versioning is enabled to ensure that the chunkHash stay consistent, but we have the same need when using hashed-asset-bundle